### PR TITLE
Fixed #1603

### LINKED ISSUE #1603: Complex names on SELECTs throw Sizzle Error
When you go to change a select / drop down with a complex name a Sizzle error is thrown. Tested in Chrome 45, 47, Safari 9.0. Using v1.14.0.

```
Uncaught Error: Syntax error, unrecognized expression: label[for='custom-data[1]['key']'], label[for='custom-data[1]['key']'] *

Stack Trace:
Sizzle.error                  @  jquery2.js:1459
Sizzle.tokenize               @  jquery2.js:2076
Sizzle.compile                @  jquery2.js:2447
Sizzle.select                 @  jquery2.js:2537
Sizzle                        @  jquery2.js:856
Sizzle.matches                @  jquery2.js:1397
jQuery.filter                 @  jquery2.js:2673
winnow                        @  jquery2.js:2656
jQuery.fn.extend.filter       @  jquery2.js:2705
$.extend.errorsFor            @  jquery.validate.js:860
$.extend.prepareElement       @  jquery.validate.js:604
$.extend.element              @  jquery.validate.js:435
$.extend.defaults.onfocusout  @  jquery.validate.js:267
delegate                      @  jquery.validate.js:381
jQuery.event.dispatch         @  jquery2.js:4436
elemData.handle               @  jquery2.js:4122
jQuery.event.trigger          @  jquery2.js:4351
jQuery.event.simulate         @  jquery2.js:4653
handler                       @  jquery2.js:4781
```

https://jsfiddle.net/matthewcallis/zbbusgve/

### PR DESCRIPTION
**Pull request by Arkni**
//cc @staabm

### CODE REVIEW

**Comment by staabm** [comment]
looks perfekt so far :)

**Comment by Arkni** [comment]
Will push the fix soon

**Comment by Arkni** [comment]
Done!

**Comment by Arkni** [comment]
@staabm 
Done!

File: src/core.js [hunk 1]
```diff
@@ -821,6 +821,7 @@ $.extend( $.validator, {
 					group = this.groups[ element.name ];
 					if ( group ) {
 						$.each( this.groups, function( name, testgroup ) {
+							name = name.replace( /'/g, "\\'" );
 							if ( testgroup === group ) {
```
Comment (staabm) [inline src/core.js]: Please move the escaping into the `if` more close to the place which requires the escaping

Comment (staabm) [inline src/core.js]: Maybe dven move it into the selector expression instead of re-assign the local var

Comment (Arkni) [inline src/core.js]: Sorry for the mess
Done!

**Comment by staabm** [comment]
Awesome thx

**Comment by Arkni** [comment]
You're welcome :)

**Pull request merged**